src/daft-core/src/array/ops/utf8.rs

```diff
@@ -787,6 +810,195 @@ impl Utf8Array {
         Ok(result)
     }
 
+    pub fn substr<I, U>(
+        &self,
+        start: &DataArray<I>,
+        length: Option<&DataArray<U>>,
+    ) -> DaftResult<Utf8Array>
+    where
+        I: DaftIntegerType,
+        <I as DaftNumericType>::Native: Ord,
+        U: DaftIntegerType,
```

Comment:
Minor nit: Is there a reason why you use `U` for this generic type? Perhaps `J` would make more sense

Reply:
No particular reason, put that on autopilot as a second variable because of <T, U>  generic types convention